nix/nim-status/build.nix

```diff
@@ -0,0 +1,126 @@
+{ pkgs, stdenv, lib, fetchFromGitHub
+# Dependencies
+, xcodeWrapper
+, git, androidPkgs
+, platform ? "android"
+, arch ? "386"
+, api ? "23" }:
+
+let
+  inherit (lib) attrNames getAttr strings concatStringsSep concatMapStrings;
+
+  osId = builtins.elemAt (builtins.split "\-" stdenv.hostPlatform.system) 2;
+  osArch = builtins.elemAt (builtins.split "\-" stdenv.hostPlatform.system) 0;
+
+  ANDROID_HOME = androidPkgs;
+  ANDROID_NDK_HOME = "${androidPkgs}/ndk-bundle";
+
+  nimIosPatch = pkgs.fetchurl { url = "https://patch-diff.githubusercontent.com/raw/nim-lang/Nim/pull/15268.patch"; sha256 = "04jg0ngqlyzwrc19rafx4s257my6bhd27myv4p78hmsyzhp36za6"; };
+  patchedNim = pkgs.nim.overrideAttrs (_: { 
+    version = "pr-15268"; 
+    patches = [ nimIosPatch ]; 
+  });
```

Comment:
Lets move this to `nix/overlay.nix` and just replace default `nim` with this one.